src/legacy/ui/public/agg_types/controls/time_interval.tsx

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { get, find } from 'lodash';
+import React, { useEffect } from 'react';
+
+import { EuiFormRow, EuiIconTip, EuiComboBox, EuiComboBoxOptionProps } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { AggParamEditorProps } from '../../vis/editors/default';
+import { AggParamOption } from '../agg_param';
+import { parseInterval } from '../utils';
+
+interface ComboBoxOption extends EuiComboBoxOptionProps {
+  key: string;
+}
+
+function TimeIntervalParamEditor({
+  agg,
+  aggParam,
+  editorConfig,
+  value,
+  setValue,
+  showValidation,
+  setValidity,
+}: AggParamEditorProps<string>) {
+  const timeBase: string = get(editorConfig, 'interval.timeBase');
+  const options = timeBase
+    ? []
+    : (aggParam.options || []).reduce(
+        (filtered: ComboBoxOption[], option: AggParamOption) => {
+          if (option.enabled ? option.enabled(agg) : true) {
```

Comment:
`if (option.enabled && option.enabled(agg))`

Reply:
Some options don't have `enabled` method and they should be displayed. The logic was moved from here https://github.com/elastic/kibana/pull/34991/files?file-filters%5B%5D=.js&hide-deleted-files=true#diff-a17d2b8a55911e4af0fdaacffbbe87afL85